profiles/ldap.py

```diff
@@ -312,27 +312,53 @@ def get_image(uid):
         account = ldap_get_member(uid)
         image = account.jpegPhoto
         github = account.github
+        twitter = account.twitterName
     except KeyError:
-        return redirect(get_gravatar("null"), code=302)
+        return redirect(get_gravatar(), code=302)
 
+    # Return stored Image
     if image:
         return image
 
-    elif github:
+    # Get Gravitar
+    url = get_gravatar(uid)
+    try:
+        gravitar = urllib.request.urlopen(url)
+        if gravitar.getcode() == 200:
+            return redirect(url, code=302)
+        else:
+            pass
+    except urllib.error.HTTPError:
+        pass
+
+    # Get GitHub Photo
+    if github:
         url = "https://github.com/" + github + ".png?size=250"
         try:
             urllib.request.urlopen(url)
             return redirect(url, code=302)
         except urllib.error.HTTPError:
-            return redirect(get_gravatar("null"), code=302)
+            pass
 
-    else:
-        return redirect(get_gravatar("null"), code=302)
+    # Get Twitter Photo
+    if twitter:
+        url = "https://twitter.com/" + twitter + "/profile_image?size=original"
+        try:
+            urllib.request.urlopen(url)
+            return redirect(url, code=302)
+        except urllib.error.HTTPError:
+            pass
 
+    # Fall back to default
+    return redirect(get_gravatar(), code=302)
```

Comment:
I don't like making 2 requests to gravatar in case they don't have an image. We should just try everything before gravatar and have it be last to serve the mystery man in case they don't have any avatars.

Reply:
So my reasoning for using gravatar first, is that there are several CSH accounts that have Gravatars as their higher quality image. When looking at some old fuck accounts, when falling to twitter or github before gravatar, they had what looked like older, squashed, or lower quality images. I think it's better to keep the expected functionality to old profiles when retrieving images

Reply:
That makes sense, so just fix the gravitar -> gravatar typo and I'll merge it